ladybug/location.py

```diff
@@ -177,28 +192,17 @@ def meridian(self):
 
     def duplicate(self):
         """Duplicate location."""
-        return Location(self.city, self.state, self.country,
-                        self.latitude, self.longitude, self.time_zone, self.elevation,
-                        self.station_id, self.source)
+        return self.__copy__()
 
-    @property
-    def ep_style_location_string(self):
```

Comment:
This is a breaking change. My suggestion is to phase it out in the next couple of releases and meanwhile add a DeprecationWarning.

Reply:
Here is an example in the core library:

https://github.com/python/cpython/blob/822922af90f389b20166f0cd6f5b7c76fd6d42d9/Lib/smtpd.py#L188-L198

Reply:
What other code is using this right now that would be broken? It's not used in ladybug_grasshopper right now and I am sending a PR to honeybee_energy that will ensure things keep working over there after the change. Is it being used for the Dynamo plugin?  I would rather make this change now before it's actually used in a stable release of a plugin and people are really relying on it.

BTW, if you are just worried about the few users we have of the pure Python libraries right now, I think these people would rather know sooner than later that this isn't supported.

Reply:
I'm also happy to change this PR to be a `refactor` rather than a `fix`, which should trigger a new breaking change release.

Reply:
> It's not used in ladybug_grasshopper right now and I am sending a PR to honeybee_energy that will ensure things keep working over there after the change. Is it being used for the Dynamo plugin?

It is about the library itself not about who else is using it. We should do it right by using DeprecationWarning and deprecate it in a couple of releases.

>  I would rather make this change now before it's actually used in a stable release of a plugin and people are really relying on it.

I didn't say we shouldn't make the change. I just think we should do it right. You can add the new one and keep the backwards compatibility for a number of releases with DeprecationWarning. That will give other developers a chance to update their code to the latest version.

Reply:
Ok, well, if the point is to practice "doing it right", we need some type of system to remind us to actually implement the breaking change because I know that I'm most likely going to forget about it otherwise. Do we want to keep the reminders as github issues (I'll likely forget to check them here unless maybe we are able to set a reminder about them through Zenhub)? Or Basecamp?

There are also some questions to figure out like "how long does the deprecation warning need to be around before we're able to implement the change?"

I was hoping we wouldn't have to worry about setting up this whole process until its really needed but, if you want to figure this out now, we can do that.

Reply:
> If it's only this particular breaking change that you are worried about in this PR, I can just take this deletion out until we can discuss these things.

Let's talk before you remove them. I feel you are overthinking this. It is a common practice and I didn't come up with it. You can just look up the best practices for code deprecation.

Reply:
I know it's a common practice and it's something that we'll definitely need to do regularly after the official release (much like the use of development branches that we aren't really using right now). But, like the use of the development branch, there are a couple of rules and best practices about them that, if not followed, make it easy to ignore.

Reply:
I removed this change from this PR and I set up an issue tag for the changes that we should probably give a deprecation warning about before the next plugin release: https://github.com/ladybug-tools/ladybug/labels/for%20the%20next%20release

I'll aim to send a "deprecation warning PR" at some point soon.